src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadTest.java

```diff
@@ -1,125 +1,217 @@
+                    // keep schema and name the same
+                    .withName("upload-test-non-json")
+                    .withFieldDefinitions(ImmutableList.of(
+                            makeUploadFieldDef("nonJson.txt", UploadFieldType.ATTACHMENT_BLOB),
+                            makeUploadFieldDef("jsonFile.json", UploadFieldType.ATTACHMENT_JSON_BLOB)))
+                    .build();
+            researcherClient.createOrUpdateUploadSchema(nonJsonSchema);
+        }
     }
 
-    @After
-    public void after() {
-        testUser.signOutAndDeleteUser();
-        deleteTextFile();
+    @AfterClass
+    public static void afterClass() {
+        if (researcher != null) {
+            try {
+                researcher.signOutAndDeleteUser();
+            } catch (RuntimeException ex) {
+                LOG.error("Error deleting temporary researcher account: " + ex.getMessage(), ex);
+            }
+        }
+
+        if (user != null) {
+            try {
+                user.signOutAndDeleteUser();
+            } catch (RuntimeException ex) {
+                LOG.error("Error deleting temporary user account: " + ex.getMessage(), ex);
+            }
+        }
     }
 
     @Test
-    public void canRequestUploadAndUpload() {
-        UploadRequest request = createRequest();
-        UploadSession session = user.requestUploadSession(request);
+    public void iosSurvey() throws Exception {
+        testUpload("ios-survey-encrypted");
+    }
 
-        user.upload(session, request, test);
+    @Test
+    public void jsonData() throws Exception {
+        testUpload("json-data-encrypted");
     }
 
     @Test
-    public void cannotUploadAfterExpirationDate() {
-        UploadRequest request = createRequest();
-        UploadSession session = user.requestUploadSession(request);
+    public void nonJson() throws Exception {
+        testUpload("non-json-encrypted");
+    }
 
-        try {
-            // Get number of milliseconds between now and expiration time, plus one second.
-            long millis = session.getExpires()
-                            .minus(DateTime.now().getMillis())
-                            .plusSeconds(1)
-                            .getMillis();
-            Thread.sleep(millis);
-        } catch (InterruptedException e) {
-            throw new BridgeSDKException(e);
+    private static void testUpload(String fileLeafName) throws Exception {
+        // set up request
+        String filePath = resolveFilePath(fileLeafName);
+        UploadRequest req = makeRequest(filePath);
+
+        // upload to server
+        UserClient userClient = user.getSession().getUserClient();
+        UploadSession session = userClient.requestUploadSession(req);
+        String uploadId = session.getId();
+        LOG.info("UploadId=" + uploadId);
+        userClient.upload(session, req, filePath);
+
+        // get validation status
+        UploadValidationStatus status = null;
```

Comment:
Not to add now, but a method that took a callback rather than requiring polling (we would do the polling, or a newer approach) might be convenient for SDK user.